### PR TITLE
feat(drawer): add new types to the Drawer

### PR DESCRIPTION
**Pull request by dmaklygin1**
[TOP-1329](https://toptal-core.atlassian.net/browse/TOP-1329)

### Description

We have to have all existing in Base UI version of Drawer: 'narrow', 'regular', 'medium', 'wide', 'ultra-wide'

⚠️ BREAKING CHANGE ⚠️: Now, `regular` type is a `narrow`, `wide` -> `regular`

### Screenshots

Picasso | Drawer 2021-06-10 14-32-57

### Review

- [ ] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [ ] Annotate all `props` in component with documentation
- [ ] Create `examples` for component
- [ ] Ensure that deployed demo has expected results and good examples
- [ ] Ensure that tests pass by running `yarn test`
- [ ] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [ ] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

File: packages/picasso-lab/src/Drawer/story/Widths.example.tsx [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
-      >
-        Show wide drawer
-      </Button>
+      <Select
+        onChange={handleOnChange}
+        options={options}
+        value={width}
+        placeholder='Choose a width...'
+        width='auto'
+      />
```
Comment (OleksandrNechai) [inline packages/picasso-lab/src/Drawer/story/Widths.example.tsx]: When I was testing, I was confused by the missing button `Show drawer`. So you see select and wondering what should I do next. Experimenting shows you have to change the value of select, but not obvious.

File: packages/picasso-lab/src/Drawer/story/Widths.example.tsx [hunk 2]
```diff
@@ -1,32 +1,34 @@
-import { Button, Container, List } from '@toptal/picasso'
+import { Container, List, Select } from '@toptal/picasso'
 import { Drawer } from '@toptal/picasso-lab'
 import React, { useState } from 'react'
 
-type WidthType = 'regular' | 'wide'
+type WidthType = 'narrow' | 'regular' | 'medium' | 'wide' | 'ultra-wide'
+
+const options = ['narrow', 'regular', 'medium', 'wide', 'ultra-wide'].map(
+  key => ({
```
Comment (OleksandrNechai) [inline packages/picasso-lab/src/Drawer/story/Widths.example.tsx]:
```suggestion
  option => ({
```

File: packages/picasso-lab/src/Drawer/story/Widths.example.tsx [hunk 3]
```diff
@@ -1,32 +1,34 @@
...
+)
 
 const Example = () => {
-  const [width, setWidth] = useState<WidthType>('regular')
+  const [width, setWidth] = useState<WidthType>('narrow')
   const [open, setOpen] = useState(false)
 
-  const handleOnButtonClick = (width: WidthType) => {
-    setWidth(width)
+  const handleOnChange = (event: React.ChangeEvent<{ value: string }>) => {
```
Comment (OleksandrNechai) [inline packages/picasso-lab/src/Drawer/story/Widths.example.tsx]: :+1:

**Review by OleksandrNechai — APPROVED** [review]
Thank you so much for contributing :bow: 
Reminder `BREAKING CHANGE: ...`

**Review by ertrzyiks — APPROVED** [review]
Perfect 👌

File: packages/picasso-lab/src/Drawer/story/Widths.example.tsx (hunk 1 above)
Comment (dmaklygin1) [reply packages/picasso-lab/src/Drawer/story/Widths.example.tsx]: I've already got it back. So now, there is a list of buttons

File: packages/picasso-lab/src/Drawer/story/index.jsx [hunk 4]
```diff
@@ -53,26 +53,25 @@ page
...
+
+        await testPage.click(buttonSelector)
+        await testPage.waitFor(contentSelector, {
+          visible: true
+        })
+        await makeScreenshot({
+          isFullScreen: true,
+          selector: contentSelector
+        })
+        await testPage.click('[role="presentation"] button')
```
[the remark this answers is not in the document]
Comment (OleksandrNechai) [reply packages/picasso-lab/src/Drawer/story/index.jsx:70]: These tests are great! I now think we might want to do it with Cypress+Happo as this is now our approach to integration testing. WDYT?

**Review by OleksandrNechai — APPROVED** [review]

File: packages/picasso-lab/src/Drawer/story/Widths.example.tsx [hunk 5]
```diff
@@ -1,32 +1,31 @@
...
+const types: WidthType[] = ['narrow', 'regular', 'medium', 'wide', 'ultra-wide']
 
 const Example = () => {
-  const [width, setWidth] = useState<WidthType>('regular')
+  const [width, setWidth] = useState<WidthType>('narrow')
   const [open, setOpen] = useState(false)
 
-  const handleOnButtonClick = (width: WidthType) => {
-    setWidth(width)
+  const handleOnButtonClick = (type: WidthType) => {
```
[the remark this answers is not in the document]
Comment (dmaklygin1) [reply packages/picasso-lab/src/Drawer/story/Widths.example.tsx:13]: hm, I thought, the `handleOn...` is specified: https://toptal-core.atlassian.net/wiki/spaces/ENG/pages/1211335036/React#Event-Handling

**Comment by dmaklygin1** [comment]
@toptal-bot run visual

**Pull request merged**